libs/application/templates/party-application/src/fields/EndorsementList/index.tsx

```diff
@@ -1,75 +1,87 @@
-  },
-  {
-    date: '21.01.2021',
-    name: 'Stefán Haukdal',
-    nationalRegistry: '1991921335',
-    address: 'Skúr hjá mömmu',
-    hasWarning: true,
-  },
-  {
-    date: '21.01.2021',
-    name: 'Brian Johannesen',
-    nationalRegistry: '1991921335',
-    address: 'Reykjavík',
-  },
-  {
-    date: '21.01.2021',
-    name: 'Örvar Þór Sigurðsson',
-    nationalRegistry: '1991921335',
-    address: 'Baugholt 15',
-  },
-  {
-    date: '21.01.2021',
-    name: 'Örvar Þór Sigurðsson',
-    nationalRegistry: '1991921335',
-    address: 'Baugholt 15',
-    hasWarning: true,
-  },
-]
+const GET_ENDORSEMENT_LIST = gql`
+  query endorsementSystemGetEndorsements($input: FindEndorsementListInput!) {
+    endorsementSystemGetEndorsements(input: $input) {
+      id
+      endorser
+      meta {
+        fullName
+        address
+      }
+      created
+      modified
+    }
+  }
+`
 
 const EndorsementList: FC<FieldBaseProps> = ({ application }) => {
   const { formatMessage } = useLocale()
-
+  const endorsementListId = (application.externalData?.createEndorsementList
+    .data as any).id
   const [searchTerm, setSearchTerm] = useState('')
-  const [signatures, setSignatures] = useState(SIGNATURES)
+  const [endorsements, setEndorsements] = useState<Endorsement[]>()
   const [showWarning, setShowWarning] = useState(false)
 
-  const namesCountString = formatMessage(
-    SIGNATURES.length > 1
-      ? formatMessage(m.endorsementList.namesCount)
-      : formatMessage(m.endorsementList.nameCount),
+  const [getEndorsementList, { loading, error }] = useLazyQuery(
+    GET_ENDORSEMENT_LIST,
+    {
+      pollInterval: 2000,
+      onCompleted: async ({ endorsementSystemGetEndorsements }) => {
+        if (!loading && endorsementSystemGetEndorsements) {
+          const hasEndorsements =
+            !error && !loading && endorsementSystemGetEndorsements?.length
+              ? endorsementSystemGetEndorsements.length > 0
+              : false
+          const mapToEndorsementList: Endorsement[] = hasEndorsements
+            ? endorsementSystemGetEndorsements.map((x: any) => ({
+                date: x.created,
+                name: x.meta.fullName,
+                nationalId: x.endorser,
+                address: x.meta.address ? x.meta.address : '',
+                hasWarning: false,
+                id: x.id,
+              }))
+            : undefined
+          setEndorsements(mapToEndorsementList)
+        }
+      },
+    },
   )
 
+  useEffect(() => {
+    getEndorsementList({
+      variables: {
+        input: {
+          listId: endorsementListId,
+        },
+      },
+    })
+  }, [])
+
+  const namesCountString = formatMessage(
+    endorsements && endorsements.length > 1
+      ? m.endorsementList.namesCount
+      : m.endorsementList.nameCount,
+  )
   return (
     <Box marginBottom={8}>
       <CopyLink
-        linkUrl="www.island.is/listabókstafur/128877634/"
+        linkUrl={window.location.href}
```

Comment:
Is this supposed to copy the current URL?

Reply:
yes

Reply:
:+1: